blockchain/chain_sync/src/chain_muxer.rs

```diff
@@ -462,7 +462,7 @@ where
                 "Validating tipset received through GossipSub failed: {}",
                 why
             );
-            return Err(why.into());
+            return Err((*why).into());
```

Comment:
You can probably box the outer error and change the `From` impl.

Reply:
Solved a bit differently